packages/go_router_builder/example/lib/all_types.dart

```diff
@@ -266,16 +266,67 @@ class UriRoute extends GoRouteData {
       );
 }
 
+class IterableRoute extends GoRouteData {
+  IterableRoute({
+    this.intIterableField,
+    this.doubleIterableField,
+    this.stringIterableField,
+    this.boolIterableField,
+    this.enumIterableField,
+    this.intListField,
+    this.doubleListField,
+    this.stringListField,
+    this.boolListField,
+    this.enumListField,
+    this.intSetField,
+    this.doubleSetField,
+    this.stringSetField,
+    this.boolSetField,
+    this.enumSetField,
+  });
+
+  final Iterable<int>? intIterableField;
```

Comment:
Would be good to modulalize this to be a class. I imagine someone would like to do this
```
class IterableRoute extends GoRouteData {
  IterableRoute({
    this.myField
  })
  final MyCustomField? myField;
}

abstract class CustomField<T> {
  Iterable<T> value;
  bool get isQueryParameter;
  String get queryParameterName;
}

class MyCustomField extend CustomField<List<int>> {
   MyCustomField();

   @override
    bool get isQueryParameter => true;
    @override
    String get queryParameterName => 'some-name' ;
}
```
The idea is to be able to choose the query parameter name.

Reply:
Hi @Skogsfrae WDYT about this? The main concerns are the query parameter names are hardcoded which I imagine people would ask for customized query parameter names in the future. Although this is not something we have to fix now, but the current API, as it is written, may not be able the extend to support this feature.